_drafts/making-backend-scalable.markdown

```diff
@@ -7,11 +7,11 @@ tags: [tango, redis, scalability]
 
 ---
 
-[Tango](https://github.com/autolab/Tango) is a stand-alone, RESTful service that Autolab uses as the back-end for autograding. Tango receives grading jobs from Autolab's front-end, adds them in a job queue, assigns them to available containers for grading and shepherds the job throuout the process. In its early days, Tango was mostly used for jobs that ran in under 5 seconds. However, over the recent semesters, Autolab has grown to host classes like Distributed Systems, Machine Learning. Some of these classes have assessments that are computationally intensive on large datasets and contains bigger files.
+[Tango](https://github.com/autolab/Tango) is a stand-alone, RESTful service that Autolab uses as the back-end for autograding. Tango receives grading jobs from Autolab's front-end, adds them in a job queue, assigns them to available containers for grading and shepherds the job throuout the process. In its early days, Tango was mostly used for jobs that ran in under 5 seconds. However, over the recent semesters, Autolab has grown to host classes like Distributed Systems, Machine Learning. Some of these classes have assessments that are computationally intensive on large datasets and contains bigger files. As we looked into how to hand larger loads, we were running into the problem of how to manage queued jobs and distribute them to different instances in our back-end. To this end, we decided to take the initial step for turning Tango into a distributed system by implementing a persistent memory using Redis and using the producer consumer model.
```

Comment:
- adds them __to__ a job queue
  - preposition usage
- shepherds the job __through__ the process
  - Typo
- Autolab has grown to host classes like Distributed Systems, Machine Learning, __and Storage Systems--classes with significantly higher compute requirements.__
  - Lists of three sound better, and joining the two sentences (either with a dash or a semicolon) will improve readability